queue_job/jobrunner/runner.py

```diff
@@ -343,6 +327,48 @@ def set_job_enqueued(self, uuid):
                 (ENQUEUED, uuid),
             )
 
+    def requeue_dead_jobs(self):
+        """
+        Set started and enqueued jobs but not locked to pending
+
+        A job is locked when it's being executed
+        When a job is killed, it releases the lock
+
+        Adding a buffer on 'date_enqueued' to check
+        that it has been enqueued for more than 10sec.
+        This prevents from requeuing jobs before they are actually started.
+
+        When Odoo shuts down normally, it waits for running jobs to finish.
+        However, when the Odoo server crashes or is otherwise force-stopped,
+        running jobs are interrupted while the runner has no chance to know
+        they have been aborted.
+        """
+        query = """
+            UPDATE
+                queue_job
+            SET
+                state='pending',
+                retry=retry+1
```

Comment:
Strictly speakingn when requeuing a job that is in `enqueued` state, it has not been tried (started) yet, so increasing the retry count in that  case is not 100% correct. We could probably ignore that case for the sake of simplicitly but it's worth a comment at least.

Reply:
I have modified this part, I have split it in 2 queries:
1. Set to 'pending' and increment `retry` if job was in `state` 'started'.
2. Set to 'pending' if job was in `state` 'enqueued'.

Reply:
Ah, another thought here. It would be nice to log job uuids dans we requeue. With info level (or maybe even warning, since this is most likely a sign that something went wrong).

Reply:
Ok, I have done:
- Query every job that are in enqueued/started but not locked and older that 10sec
- Set jobs that are started to pending and increment retry + logging
- Set jobs that are enqueued to pending + logging